opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java

```diff
@@ -407,26 +409,42 @@ private static JSONObject getCheckBoxJsonObjects(Event event, JSONObject parentO
 
     public static void addObservation(Event e, JSONObject jsonObject) {
         String value = getString(jsonObject, VALUE);
-        if (StringUtils.isBlank(value)) { return; }
+        if (StringUtils.isBlank(value)) {
+            return;
+        }
 
         String type = getString(jsonObject, AllConstants.TYPE);
         if (AllConstants.CHECK_BOX.equals(type)) {
             try {
                 List<Object> optionValues = new ArrayList<>();
+                List<Object> optionEntityIds = new ArrayList<>();
                 Map<String, Object> optionKeyVals = new HashMap<>();
                 if (jsonObject.has(AllConstants.OPTIONS)) {
                     JSONArray options = jsonObject.getJSONArray(AllConstants.OPTIONS);
+                    String fieldsOpenmrsEntityId = jsonObject.optString(OPENMRS_ENTITY_ID);
+                    String fieldOpenmrsEntityParent = jsonObject.optString(OPENMRS_ENTITY_PARENT);
+                    String fieldKey = jsonObject.optString(KEY);
+                    boolean shouldBeCombined = jsonObject.optBoolean(AllConstants.COMBINE_CHECKBOX_OPTION_VALUES);
+                    String entity = getString(jsonObject, OPENMRS_ENTITY);
                     for (int i = 0; i < options.length(); i++) {
                         JSONObject option = options.getJSONObject(i);
                         boolean optionValue = option.optBoolean(VALUE);
                         if (!optionValue) {
                             continue;
                         }
-                        if (CONCEPT.equals(getString(jsonObject, OPENMRS_ENTITY))) {
+                        if (CONCEPT.equals(entity)) {
                             // For options with concepts create an observation for each
                             option.put(AllConstants.TYPE, type);
-                            option.put(AllConstants.PARENT_ENTITY_ID, jsonObject.getString(OPENMRS_ENTITY_ID));
+                            option.put(AllConstants.PARENT_ENTITY_ID, fieldsOpenmrsEntityId);
                             option.put(KEY, jsonObject.getString(KEY));
```

Comment:
`fieldKey`?

Reply:
`jsonObject.getString(OPENMRS_ENTITY_ID)` is `fieldsOpenmrsEntityId` otherwise i dont get the question

Reply:
Replace `jsonObject.getString(KEY)` with `fieldKey`.

Reply:
cool changing